src/ctapipe/calib/camera/calibrator.py

```diff
@@ -282,6 +292,44 @@ def _calibrate_dl1(self, event, tel_id):
         # store the results in the event structure
         event.dl1.tel[tel_id] = dl1
 
+    def _process_calib(self, event, tel_id):
+        waveforms = event.r1.tel[tel_id].waveform
+        if self._check_r1_empty(waveforms):
+            return
+
+        if self.image_extractor_type.tel[tel_id] not in [
+            "LocalPeakWindowSum",
+            "FixedWindowSum",
+        ]:
+            raise ValueError(
+                f"Invalid ImageExtractor '{self.image_extractor_type.tel[tel_id]}'."
+                "Valid options: LocalPeakWindowSum FixedWindowSum."
+            )
+
+        n_channels, n_pixels, n_samples = waveforms.shape
+
+        broken_pixels = event.mon.tel[tel_id].pixel_status.hardware_failing_pixels
+        if broken_pixels is None:
+            broken_pixels = np.zeros(n_pixels, dtype=bool)
+
+        # TODO: See #1836 (process all provided gains in the ImageExtractor)
+        no_gain_selection = np.zeros((n_channels, n_pixels), dtype=np.int64)
+
+        if n_channels == 2:
+            no_gain_selection[1] = 1
+
+        # Extract charge and arrival time and return the DL1CameraContainer
+        extractor = self.image_extractors[self.image_extractor_type.tel[tel_id]]
+
+        dl1 = extractor(
+            waveforms,
+            tel_id=tel_id,
+            selected_gain_channel=no_gain_selection,
```

Comment:
Can you please explain the logic behind this? I see that if n_channels is 1, you pass 1d array of zeros, and if n_channel is 2 it is a 2d array of zeros and ones...

Reply:
I think it's just a workaround because the ImageExtractors are currently designed to work with gainselected images. Therefore this issue should be addressed in a separate PR.

Reply:
I think we should just add an API to the extractors for not gain selected waveforms. 

I.e. allow `selected_gain_channel=None` which should mean no gain selection happened and that the waveform has shape `n_channels, n_pixels, n_samples`

Reply:
yes, I'm trying to solve it right now. I will let you know.

Reply:
Im also currently working on #1836 which would solve it - ill create a draft PR today or tomorrow, maybe we can discuss it there.

Reply:
great! thank you @Hckjs I have also contacted you on the CTA North slack channel with the changes I had in mind so far. Maybe they can be helpful.